Assets/PatchKit Patcher/Scripts/Debug/DebugMenu.cs

```diff
@@ -0,0 +1,242 @@
+        }
+
+        void OnGUI()
+        {
+            if (Event.current.Equals(Event.KeyboardEvent(KeyCode.D.ToString())))
+            {
+                if (_showDebugMenu)
+                {
+                    Close();
+                }
+                else
+                {
+                    Open();
+                }
+            }
+
+            if (_showDebugMenu)
+            {
+                GUI.DrawTexture(_rect, Texture2D.whiteTexture);
+                GUI.Window(0, _rect, Draw, "Debug Menu");
+            }
+            else if (_showPopup)
+            {
+                GUI.DrawTexture(_popupRect, Texture2D.whiteTexture);
+                GUI.Window(1, _popupRect, DrawPopup, "Information");
+            }
+        }
+
+        void Draw(int id)
+        {
+            if (GUILayout.Button("Open Patcher log file"))
+            {
+                OpenPatcherLogFile();
+            }
+
+            if (GUILayout.Button("Open Patcher log file location"))
+            {
+                OpenPatcherLogFileLocation();
+            }
+
+            if (GUILayout.Button("Open Launcher log file"))
+            {
+                OpenLauncherLogFile();
+            }
+
+            if (GUILayout.Button("Open Launcher log file location"))
+            {
+                OpenLauncherLogFileLocation();
+            }
+
+            if (Patcher.Instance.IsAppInstalled.Value)
+            {
+                if (GUILayout.Button("Verify all app files"))
+                {
+                    VerifyAllAppFiles();
+                }
+
+                if (GUILayout.Button("Remove all app files"))
+                {
+                    RemoveAllAppFiles();
+                }
+            }
+        }
+
+        private void VerifyAllAppFiles()
+        {
+            StartCoroutine(SetUserDecision(Patcher.UserDecision.VerifyFiles));
+        }
+
+        private void RemoveAllAppFiles()
+        {
+            StartCoroutine(SetUserDecision(Patcher.UserDecision.UninstallApp));
+        }
+
+        private IEnumerator SetUserDecision(Patcher.UserDecision userDecision)
+        {
+            while (Patcher.Instance.State.Value != PatcherState.WaitingForUserDecision)
+            {
+                yield return new WaitForSeconds(0.5f);
+            }
+
+            Patcher.Instance.SetUserDecision(userDecision);
+
+            Close();
+        }
+
+        void OpenFile(string path)
+        {
+            Close();
+            try
+            {
+                Process.Start(path);
+            }
+            catch (Exception e)
+            {
+                _graphicRaycaster.enabled = false;
+                OpenPopup(string.Format("The directory/file cannot be found: {0}", path));
+            }
+        }
+
+        private void OpenPopup(string popupMessage)
+        {
+            _popupMessage = popupMessage;
+            DebugLogger.LogError(popupMessage);
+            _showDebugMenu = false;
```

Comment:
We don't want to hide the debug menu, as the popup is a part of it.

Reply:
done